src/App.tsx

```diff
@@ -1,14 +1,80 @@
-/* eslint-disable max-len */
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { debounce } from './utils/debounce';
 
 export const App: React.FC = () => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [searchQuery, setSearchQuery] = useState('');
+  const [filterName, setFilterName] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+
+  useEffect(() => {
+    setIsLoading(true);
+    getTodos().then(todosFromServer => {
+      setIsLoading(false);
+      setTodos(todosFromServer);
+    });
+  }, []);
+
+  const applyQuery = useCallback(debounce(setAppliedQuery, 300), []);
+
+  function handleToggleEye(todoId: number | null) {
+    if (todoId) {
+      setSelectedTodo(todos.find(({ id }) => id === todoId) || null);
+    } else {
+      setSelectedTodo(null);
+    }
+  }
```

Comment:
- This function is not just toggle eye icon
- Probably, you can just pass the todo itself?